django/thunderstore/repository/migrations/0029_create_namespaces_for_teams.py

```diff
@@ -0,0 +1,23 @@
+# Generated by Django 3.1.7 on 2021-11-04 20:32
+
+from django.db import migrations
+
+
+def forwards(apps, schema_editor):
+    Namespace = apps.get_model("repository", "Namespace")
+    Team = apps.get_model("repository", "Team")
+    teams = Team.objects.all().order_by("id")
+    new_namespaces = [Namespace(team=team, name=team.name) for team in teams]
```

Comment:
This is still loading everything to memory, which somewhat defeats the purpose of doing batch inserts. I don't think it's a problem in practice with our object count, but just thought I'd note regardless for future reference.

When enumerating over potentially large sets of data, you'd want to use the [QuerySet.iterator()](https://docs.djangoproject.com/en/3.2/ref/models/querysets/#iterator) method to avoid loading everything to memory (which usually gets you OOM killed).

Bulk inserts are also good for inserting efficiently, a bulk of 2000 is decent, so ideally you'd probably want something like
```python
T = TypeVar("T")


def batch(batch_size: int, iterable: Iterable[T]) -> Iterable[List[T]]:
    collected = []
    for entry in iterable:
        collected.append(entry)
        if len(collected) >= batch_size:
            yield collected
            collected = []
    if len(collected) > 0:
        yield collected


def forwards(apps, schema_editor):
    Namespace = apps.get_model("repository", "Namespace")
    Team = apps.get_model("repository", "Team")

    for entry in batch(2000, map(lambda x: Namespace(team=x, name=x.name), Team.objects.all())):
        Namespace.objects.bulk_create(entry)
```

In the example above, you only ever load up to a maximum of 2000 objects into memory at once. Could further be optimized into only fetching the data you need (e.g. `.values("id", "name")` in this case)

Reply:
The example seems to work nicely. Added the values thingie